core/src/banking_simulation.rs

```diff
@@ -0,0 +1,908 @@
+            blockstore.get_new_shred_signal(0),
+            &leader_schedule_cache,
+            &genesis_config.poh_config,
+            None,
+            exit.clone(),
+        );
+        let poh_recorder = Arc::new(RwLock::new(poh_recorder));
+        let poh_service = PohService::new(
+            poh_recorder.clone(),
+            &genesis_config.poh_config,
+            exit.clone(),
+            bank.ticks_per_slot(),
+            DEFAULT_PINNED_CPU_CORE,
+            DEFAULT_HASHES_PER_BATCH,
+            record_receiver,
+        );
+
+        // Enable BankingTracer to approximate the real environment as close as possible because
+        // it's not expected to disable BankingTracer on production environments.
+        //
+        // It's not likely for it to affect the banking stage performance noticeably. So, make sure
+        // that assumption is held here. That said, it incurs additional channel sending,
+        // SystemTime::now() and buffered seq IO, and indirectly functions as a background dropper
+        // of `BankingPacketBatch`.
+        //
+        // Lastly, the actual retraced events can be used to evaluate simulation timing accuracy in
+        // the future.
+        let (retracer, retracer_thread) = BankingTracer::new(Some((
+            &blockstore.banking_retracer_path(),
+            exit.clone(),
+            BANKING_TRACE_DIR_DEFAULT_BYTE_LIMIT,
+        )))
+        .unwrap();
+        assert!(retracer.is_enabled());
+        info!(
+            "Enabled banking retracer (dir_byte_limit: {})",
+            BANKING_TRACE_DIR_DEFAULT_BYTE_LIMIT,
+        );
+
+        let (non_vote_sender, non_vote_receiver) = retracer.create_channel_non_vote();
+        let (tpu_vote_sender, tpu_vote_receiver) = retracer.create_channel_tpu_vote();
+        let (gossip_vote_sender, gossip_vote_receiver) = retracer.create_channel_gossip_vote();
+
+        let connection_cache = Arc::new(ConnectionCache::new("connection_cache_sim"));
+        let (replay_vote_sender, _replay_vote_receiver) = unbounded();
+        let (retransmit_slots_sender, retransmit_slots_receiver) = unbounded();
+        let shred_version = compute_shred_version(
+            &genesis_config.hash(),
+            Some(&bank_forks.read().unwrap().root_bank().hard_forks()),
+        );
+        let (sender, _receiver) = tokio::sync::mpsc::channel(1);
+
+        // Create a completely-dummy ClusterInfo for the broadcast stage.
+        // We only need it to write shreds into the blockstore and it seems given ClusterInfo is
+        // irrelevant for the neccesary minimum work for this simulation.
+        let random_keypair = Arc::new(Keypair::new());
+        let cluster_info = Arc::new(ClusterInfo::new(
+            Node::new_localhost_with_pubkey(&random_keypair.pubkey()).info,
+            random_keypair,
+            SocketAddrSpace::Unspecified,
+        ));
+        let broadcast_stage = BroadcastStageType::Standard.new_broadcast_stage(
```

Comment:
is there not a "disabled" broadcast stage we could use?

Reply:
seems not? anyway, it's effectively disabled by passing virtually empty `ClusterInfo`... 

btw added a comment: 2f907584e60b780716bb38dbafc0828867954d2a

Reply:
Got it - yeah definitely outside the scope. Need some sort of "BroadcastStage" but ideally we'd not need `ClusterInfo`, and skip attempting broadcast just forwarding shreds to blockstore. The reason we need it now is due to the interface of `BroadcastStage`.